lib/datadog/core/telemetry/logger.rb

```diff
@@ -25,7 +25,17 @@ def error(description)
           private
 
           def instance
-            Datadog.send(:components).telemetry
+            # `allow_initialization: false` to prevent deadlock from components lifecycle
+            components = Datadog.send(:components, allow_initialization: false)
+
+            if components && components.telemetry
+              components.telemetry
+            else
+              Datadog.logger.error(
+                'Fail to send telemetry log before components initialization or within components lifecycle'
+              )
```

Comment:
This will be seen by customers -- do we want to perhaps lower this to warn or even lower? It's not like they'll be able to do anything about this issue, since this is a bug on our side